web/src/message_notifications.js

```diff
@@ -0,0 +1,447 @@
+import $ from "jquery";
+
+import * as alert_words from "./alert_words";
+import * as blueslip from "./blueslip";
+import * as compose_banner from "./compose_banner";
+import {$t} from "./i18n";
+import * as message_lists from "./message_lists";
+import * as message_parser from "./message_parser";
+import * as narrow from "./narrow";
+import * as notifications from "./notifications";
+import * as people from "./people";
+import * as spoilers from "./spoilers";
+import * as stream_data from "./stream_data";
+import * as ui_util from "./ui_util";
+import * as unread from "./unread";
+import {user_settings} from "./user_settings";
+import * as user_topics from "./user_topics";
+
+export function initialize({on_click_scroll_to_selected}) {
+    $(window).on("focus", () => {
+        for (const notice_mem_entry of notifications.notice_memory.values()) {
+            notice_mem_entry.obj.close();
+        }
+        notifications.notice_memory.clear();
+    });
```

Comment:
It might make sense for this on-focus handler to stay in `notification.js`, since it seems like a low-level data clearing thing separate from all business logic about what the notifications are?

Reply:
Sure.